saleor/checkout/complete_checkout.py

```diff
@@ -461,10 +463,17 @@ def _create_order_line_discount_object_for_voucher(
     code = checkout_line_info.voucher_code
     discount_reason = f"Voucher code: {code}"
 
+    if channel.use_legacy_line_voucher_propagation_for_order:
+        # Previously voucher discount was always set as fixed value. To keep it in the
+        # same way as previously we force the fixed type when legacy flow is used
+        value_type = DiscountValueType.FIXED
+    else:
+        value_type = voucher.discount_value_type
+
     return OrderLineDiscount(
         line=order_line,
         type=DiscountType.VOUCHER,
-        value_type=DiscountValueType.FIXED,
+        value_type=value_type,
         value=discount_amount,
```

Comment:
TODO: If you change the discount type to `percentage` the value should be properly changes as well, it should shows the percentage value discount applied

Reply:
Thanks Iga. I totally missed that